### PR TITLE
[zh-CN]: Update translation of X-Forwarded-For

### PR DESCRIPTION
**Pull request by dushaoshuai**
### Description

Such as the title.

### Motivation

Ditto.

### Additional details

Nope.

### Related issues and pull requests
Nope.

### CODE REVIEW

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 1]
```diff
@@ -5,53 +5,103 @@ slug: Web/HTTP/Headers/X-Forwarded-For
 
 {{HTTPSidebar}}
 
-**`X-Forwarded-For`** (XFF) 在客户端访问服务器的过程中如果需要经过 HTTP 代理或者负载均衡服务器，可以被用来获取最初发起请求的客户端的 IP 地址，这个消息首部成为事实上的标准。在消息流从客户端流向服务器的过程中被拦截的情况下，服务器端的访问日志只能记录代理服务器或者负载均衡服务器的 IP 地址。如果想要获得最初发起请求的客户端的 IP 地址的话，那么 X-Forwarded-For 就派上了用场。
+当客户端通过代理服务器连接到 web 服务器时，**`X-Forwarded-For`**（XFF）请求首部可用于标识客户端的原始 IP 地址。该首部是事实上的标准。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:8]:
```suggestion
**`X-Forwarded-For`**（XFF）请求标头是一个事实上的标准请求头，用于识别通过代理服务器连接到 Web 服务器到客户端的原始 IP 地址。
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 2]
```diff
@@ -5,53 +5,103 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
 
-这个消息首部会被用来进行调试和统计，以及生成基于位置的定制化内容，按照设计的目的，它会暴露一定的隐私和敏感信息，比如客户端的 IP 地址。所以在应用此消息首部的时候，需要将用户的隐私问题考虑在内。
+> **警告：** 对此首部的不恰当使用可能带来安全风险。有关详细信息，请参阅[安全和隐私考虑](#安全和隐私考虑)。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:10]:
```suggestion
> **警告：** 对此标头的不恰当使用可能带来安全风险。有关详细信息，请参阅[安全和隐私考虑](#安全和隐私考虑)。
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 3]
```diff
@@ -5,53 +5,103 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
 
-HTTP 协议中的 {{HTTPHeader("Forwarded")}} 是这个消息首部的标准化版本。
+当客户端直接连接到服务器时，其 IP
+地址被发送给服务器（并且经常被记录在服务器的访问日志中）。但是如果客户端通过[正向或反向代理服务器](https://en.wikipedia.org/wiki/Proxy_server)进行连接，服务器就只能看到最后一个代理服务器的 IP
+地址，这个 IP 通常没什么用。如果最后一个代理服务器是与服务器安装在同一台主机上的负载均衡服务器，则更是如此。`X-Forwarded-For` 的出现，就是为了向服务器提供更有用的客户端 IP 地址。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:14]:
```suggestion
当客户端直接连接到服务器时，其 IP 地址被发送给服务器（并且经常被记录在服务器的访问日志中）。但是如果客户端通过[正向或反向代理服务器](https://zh.wikipedia.org/wiki/代理服务器)进行连接，服务器就只能看到最后一个代理服务器的 IP 地址，这个 IP 通常没什么用。如果最后一个代理服务器是与服务器安装在同一台主机上的负载均衡服务器，则更是如此。`X-Forwarded-For` 的出现，就是为了向服务器提供更有用的客户端 IP 地址。
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 4]
```diff
@@ -5,53 +5,103 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
 
-`X-Forwarded-For` 也是一个电子邮件相关协议中用到的首部，用来表示一封电子邮件是从其他账户转发过来的。
+有关使用此首部的详细指引，请参阅[解析](#解析)和[选择一个 IP 地址](#选择一个_ip_地址)部分。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:16]: 术语表中 header 一般译为标头，参见 [http-header](zh-CN/docs/Glossary/HTTP_header)，其他相同问题我就不一一标注了
```suggestion
有关使用此标头的详细指引，请参阅[解析](#解析)和[选择一个 IP 地址](#选择一个_ip_地址)部分。
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 5]
```diff
@@ -5,53 +5,103 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
+  </tbody>
+</table>
+
+此首部的标准化版本是 HTTP {{HTTPHeader("Forwarded")}} 首部。
+
+## 安全和隐私考虑
+
+按照设计的目的，此首部暴露隐私敏感信息，比如客户端的 IP 地址。因此，在应用此首部时，需要将用户的隐私问题考虑在内。
+
+当在客户端和服务器之间没有受信任的反向代理服务器时（比如一个负载均衡服务器），`X-Forwarded-For` 首部是不可信的。如果客户端和所有的代理服务器都没有恶意且行为良好，那么此首部中的 IP 地址列表具有[指令](#指令)部分所描述的意义。但是如果客户端或任意代理服务器存有恶意或配置错误，那么此首部的任意部分（或整体）可能是仿冒的（并且有可能不是一个列表或者甚至不包含 IP 地址）。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:37]: Ditto.

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 6]
```diff
@@ -5,53 +5,103 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
+
+如果客户端和服务器之间存在任意受信任的反向代理服务器，最终 `X-Forwarded-For` 中的 IP 地址列表是可信的，因为这些 IP 是由受信任的代理服务器添加的（只要服务器只能通过这些代理服务器进行访问而不是可以被直接访问）。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:39]:
```suggestion
如果客户端和服务器之间存在任意受信任的反向代理服务器，最终 `X-Forwarded-For` 中的 IP 地址列表是可信的，因为这些 IP 是由受信任的代理服务器添加的（*只要*服务器只能通过这些代理服务器进行访问而不是可以被直接访问）。
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 7]
```diff
@@ -5,53 +5,103 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
+
+使用 `X-Forwarded-For` 进行安全相关的操作时（比如速率限制或基于 IP 的访问控制），必须仅使用由受信任的代理服务器添加的 IP 地址。使用不可信的 IP，可能导致速率限制被规避、访问控制被绕过、内存耗尽、或其他与安全性或可用性相关的负面后果。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:41]:
```suggestion
使用 `X-Forwarded-For` 进行安全相关的操作时（比如速率限制或基于 IP 的访问控制），*必须仅*使用由受信任的代理服务器添加的 IP 地址。使用不可信的 IP，可能导致速率限制被规避、访问控制被绕过、内存耗尽、或其他与安全性或可用性相关的负面后果。
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 8]
```diff
@@ -5,53 +5,103 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
+当选择离客户端最近的 `X-Forwarded-For` IP 地址时（不可信并且不用于安全相关的目地），应该选择最左边的、第一个有效且不是私有/内部地址的 IP 地址。（要求”有效“是因为仿冒的值可能根本就不是 IP 地址；要求“不是私有/内部地址”是因为客户端使用的代理服务器可能在他们的内部网络中，在这种情况下代理服务器可能添加了[私有 IP 地址空间](https://en.wikipedia.org/wiki/Private_network)中的地址。）
+
+当选择第一个可信的 `X-Forwarded-For` 客户端 IP 地址时，需要进行额外的配置。有两种常用的方法：
+
+- **受信任代理服务器数量**：配置了互联网和服务器之间的反向代理服务器数量。从 `X-Forwarded-For` IP 列表的最右边开始搜索，第（受信任代理服务器数量 - 1）个地址就是目标地址。（例如，如果只有一个反向代理服务器，这个代理服务器会添加客户端的 IP 地址，因此应该使用最右边的地址。如果有三个反向代理服务器，最后两个 IP 地址将是内部地址。）
+- **受信任代理服务器列表**：配置了受信任反向代理服务器的 IP 或 IP 范围。从 `X-Forwarded-For` IP 列表的最右边开始搜索，跳过受信任代理服务器列表中的所有地址。第一个不匹配的地址就是目标地址。
+
+第一个可信的 `X-Forwarded-For` IP 地址可能属于一个不受信任的中间代理服务器，而不是实际的客户计算机，但这是唯一一个适用于安全用途的 IP 了。
+
+注意如果从互联网可直接连接到服务器——即使服务器也位于一个受信任的反向代理服务器之后——`X-Forwarded-For` IP 列表中的任何部分都不能被认为是可信赖的，或者可安全地用于安全相关的用途。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:98]:
```suggestion
注意如果从互联网可直接连接到服务器——即使服务器也位于一个受信任的反向代理服务器之后——`X-Forwarded-For` IP 列表中的*任何部分*都不能被认为是可信赖的，或者可安全地用于安全相关的用途。
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 9]
```diff
@@ -5,53 +5,103 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
+
+（一些反向代理服务器会自动将多个 `X-Forwarded-For` 首部合并成一个，但是最安全的做法是不要假设就是这种情况。）
+
+## 选择一个 IP 地址
+
+选择地址时，必须使用从所有 `X-Forwarded-For` 首部生成的完整 IP 列表。
+
+当选择离客户端最近的 `X-Forwarded-For` IP 地址时（不可信并且不用于安全相关的目地），应该选择最左边的、第一个有效且不是私有/内部地址的 IP 地址。（要求”有效“是因为仿冒的值可能根本就不是 IP 地址；要求“不是私有/内部地址”是因为客户端使用的代理服务器可能在他们的内部网络中，在这种情况下代理服务器可能添加了[私有 IP 地址空间](https://en.wikipedia.org/wiki/Private_network)中的地址。）
+
+当选择第一个可信的 `X-Forwarded-For` 客户端 IP 地址时，需要进行额外的配置。有两种常用的方法：
```
Comment (JasonLamv-t) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:91]:
```suggestion
当选择第一个*可信的* `X-Forwarded-For` 客户端 IP 地址时，需要进行额外的配置。有两种常用的方法：
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 10]
```diff
@@ -5,53 +5,103 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
+
+只使用多个 `X-Forwarded-For` 首部中的一个是不够的。
+
+（一些反向代理服务器会自动将多个 `X-Forwarded-For` 首部合并成一个，但是最安全的做法是不要假设就是这种情况。）
+
+## 选择一个 IP 地址
+
+选择地址时，必须使用从所有 `X-Forwarded-For` 首部生成的完整 IP 列表。
+
+当选择离客户端最近的 `X-Forwarded-For` IP 地址时（不可信并且不用于安全相关的目地），应该选择最左边的、第一个有效且不是私有/内部地址的 IP 地址。（要求”有效“是因为仿冒的值可能根本就不是 IP 地址；要求“不是私有/内部地址”是因为客户端使用的代理服务器可能在他们的内部网络中，在这种情况下代理服务器可能添加了[私有 IP 地址空间](https://en.wikipedia.org/wiki/Private_network)中的地址。）
```
Comment (JasonLamv-t) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:89]:
```suggestion
当选择离客户端最近的 `X-Forwarded-For` IP 地址时（不可信并且*不用于*安全相关的目地），应该选择最左边的、第一个*有效且不是私有/内部地址*的 IP 地址。（要求”有效“是因为仿冒的值可能根本就不是 IP 地址；要求“不是私有/内部地址”是因为客户端使用的代理服务器可能在他们的内部网络中，在这种情况下代理服务器可能添加了[私有 IP 地址空间](https://zh.wikipedia.org/wiki/专用网络)中的地址。）
```

**Review by JasonLamv-t — COMMENTED** [review]
Some suggestions.

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md (hunk 1 above)
Comment (dushaoshuai) [reply files/zh-cn/web/http/headers/x-forwarded-for/index.md:10]: 这里有个文字错误，因此我在这个 commit 里改了 https://github.com/mdn/translated-content/pull/13336/commits/9579cc589eea6b803662a82e085ebcabb74aa0d2

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md (hunk 4 above)
Comment (dushaoshuai) [reply files/zh-cn/web/http/headers/x-forwarded-for/index.md]: Done. 
[batch replace 首部 with 标头](https://github.com/mdn/translated-content/pull/13336/commits/948d5dad26fb93699bc80d85c988428dd9f39e0d).

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md (hunk 5 above)
Comment (dushaoshuai) [reply files/zh-cn/web/http/headers/x-forwarded-for/index.md]: Done. 
[batch replace 首部 with 标头](https://github.com/mdn/translated-content/pull/13336/commits/948d5dad26fb93699bc80d85c988428dd9f39e0d).

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md (hunk 6 above)
Comment (dushaoshuai) [reply files/zh-cn/web/http/headers/x-forwarded-for/index.md]: 斜体的应该是*只能*吧，[mark 只能 as italic](https://github.com/mdn/translated-content/pull/13336/commits/e7c0cb59ab8dbe2877e32d2298f804e066dfee28)

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 11]
```diff
@@ -5,53 +5,101 @@ slug: Web/HTTP/Headers/X-Forwarded-For
 
 {{HTTPSidebar}}
 
-**`X-Forwarded-For`** (XFF) 在客户端访问服务器的过程中如果需要经过 HTTP 代理或者负载均衡服务器，可以被用来获取最初发起请求的客户端的 IP 地址，这个消息首部成为事实上的标准。在消息流从客户端流向服务器的过程中被拦截的情况下，服务器端的访问日志只能记录代理服务器或者负载均衡服务器的 IP 地址。如果想要获得最初发起请求的客户端的 IP 地址的话，那么 X-Forwarded-For 就派上了用场。
+**`X-Forwarded-For`**（XFF）请求标头是一个事实上的标准请求头，用于识别通过代理服务器连接到 Web 服务器的客户端的原始 IP 地址。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:8]:
```suggestion
**`X-Forwarded-For`**（XFF）请求标头是一个事实上的用于标识通过代理服务器连接到 web 服务器的客户端的原始 IP 地址的标头。
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 12]
```diff
@@ -5,53 +5,101 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
+有关使用此标头的详细指引，请参阅[解析](#解析)和[选择一个 IP 地址](#选择一个_ip_地址)部分。
 
-| Header type                                      | {{Glossary("Request header")}} |
-| ------------------------------------------------ | ---------------------------------------- |
-| {{Glossary("Forbidden header name")}} | no                                       |
+<table class="properties">
+  <tbody>
+    <tr>
+      <th scope="row">Header type</th>
+      <td>{{Glossary("Request header")}}</td>
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:20]:
```suggestion
      <th scope="row">标头类型</th>
      <td>{{Glossary("Request header", "请求标头")}}</td>
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 13]
```diff
@@ -5,53 +5,101 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
+    </tr>
+    <tr>
+      <th scope="row">{{Glossary("Forbidden header name")}}</th>
+      <td>no</td>
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:24]:
```suggestion
      <th scope="row">{{Glossary("Forbidden header name", "禁止修改的标头")}}</th>
      <td>否</td>
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 14]
```diff
@@ -5,53 +5,101 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
+  </tbody>
+</table>
+
+此标头的标准化版本是 HTTP {{HTTPHeader("Forwarded")}} 标头。
+
+## 安全和隐私考虑
+
+按照设计的目的，此标头暴露隐私敏感信息，比如客户端的 IP 地址。因此，在应用此标头时，需要将用户的隐私问题考虑在内。
+
+当在客户端和服务器之间没有受信任的反向代理服务器时（比如一个负载均衡服务器），`X-Forwarded-For` 标头是不可信的。如果客户端和所有的代理服务器都没有恶意且行为良好，那么此标头中的 IP 地址列表具有[指令](#指令)部分所描述的意义。但是如果客户端或任意代理服务器存有恶意或配置错误，那么此标头的任意部分（或整体）可能是仿冒的（并且有可能不是一个列表或者甚至不包含 IP 地址）。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:35]: 仿冒好像主要指产品，这里用伪造可能更合适

```suggestion
当在客户端和服务器之间没有受信任的反向代理服务器时（比如一个负载均衡服务器），`X-Forwarded-For` 标头是不可信的。如果客户端和所有的代理服务器都没有恶意且行为良好，那么此标头中的 IP 地址列表具有[指令](#指令)部分所描述的意义。但是如果客户端或任意代理服务器存有恶意或配置错误，那么此标头的任意部分（或整体）可能是伪造的（并且有可能不是一个列表或者甚至不包含 IP 地址）。
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 15]
```diff
@@ -5,53 +5,101 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
+
+如果客户端和服务器之间存在任意受信任的反向代理服务器，最终 `X-Forwarded-For` 中的 IP 地址列表是可信的，因为这些 IP 是由受信任的代理服务器添加的（只要服务器*只能*通过这些代理服务器进行访问而不是可以被直接访问）。
+
+使用 `X-Forwarded-For` 进行安全相关的操作时（比如速率限制或基于 IP 的访问控制），*必须仅*使用由受信任的代理服务器添加的 IP 地址。使用不可信的 IP，可能导致速率限制被规避、访问控制被绕过、内存耗尽、或其他与安全性或可用性相关的负面后果。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:39]:
```suggestion
使用 `X-Forwarded-For` 进行安全相关的操作时（比如速率限制或基于 IP 的访问控制），*必须仅*使用由受信任的代理服务器添加的 IP 地址。使用不可信的 IP，可能导致速率限制被规避、访问控制被绕过、内存耗尽，或其他与安全性或可用性相关的负面后果。
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 16]
```diff
@@ -5,53 +5,101 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
+
+反过来，只有当使用仿冒的值不会导致负面影响时，最左边的（不可信的）值才能被使用。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:41]:
```suggestion
反过来，只有当使用伪造的值不会导致负面影响时，最左边的（不可信的）值才能被使用。
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 17]
```diff
@@ -5,53 +5,101 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
 ```
 
-其他非标准形式：
+## 解析
 
-```plain
-# Used for some Google services
-X-ProxyUser-Ip: 203.0.113.19
-```
+对 `X-Forwarded-For` 标头的不当解析可能导致仿冒的值被用于安全相关的目地，造成本文之前提及过的负面后果。
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:72]:
```suggestion
对 `X-Forwarded-For` 标头的不当解析可能导致仿冒的值被用于安全相关的目地，造成上文提及过的负面后果。
```

File: files/zh-cn/web/http/headers/x-forwarded-for/index.md [hunk 18]
```diff
@@ -5,53 +5,101 @@ slug: Web/HTTP/Headers/X-Forwarded-For
...
+
+只使用多个 `X-Forwarded-For` 标头中的一个是不够的。
+
+（一些反向代理服务器会自动将多个 `X-Forwarded-For` 标头合并成一个，但是最安全的做法是不要假设就是这种情况。）
+
+## 选择一个 IP 地址
+
+选择地址时，必须使用从所有 `X-Forwarded-For` 标头生成的完整 IP 列表。
+
+当选择离客户端最近的 `X-Forwarded-For` IP 地址时（不可信并且*不用于*安全相关的目地），应该选择最左边的、第一个*有效且不是私有/内部地址*的 IP 地址。（要求”有效“是因为仿冒的值可能根本就不是 IP 地址；要求“不是私有/内部地址”是因为客户端使用的代理服务器可能在他们的内部网络中，在这种情况下代理服务器可能添加了[私有 IP 地址空间](https://zh.wikipedia.org/wiki/专用网络)中的地址。）
```
Comment (yin1999) [inline files/zh-cn/web/http/headers/x-forwarded-for/index.md:87]:
```suggestion
当选择离客户端最近的 `X-Forwarded-For` IP 地址时（不可信并且*不用于*安全相关的目地），应该选择最左边的、第一个*有效且不是私有/内部地址*的 IP 地址。（要求”有效“是因为仿冒的值可能根本就不是 IP 地址；要求“不是私有/内部地址”是因为客户端使用的代理服务器可能在它们的内部网络中，在这种情况下代理服务器可能添加了[私有 IP 地址空间](https://zh.wikipedia.org/wiki/专用网络)中的地址。）
```

**Review by yin1999 — COMMENTED** [review]
Some suggestions ;)

**Review by yin1999 — APPROVED** [review]
Thanks @dushaoshuai. Welcome aboard :tada: :)

**Pull request merged**